### PR TITLE
feat: rename and extend deployed.yaml config [fixes DXJ-343]

### PR DESCRIPTION
**Pull request by shamsartem**
BREAKING CHANGE: renamed deployed.yaml to workers.yaml, renamed deals.aqua to workers.aqua.

* workers.yaml now has separate keys for deals and hosts just like fluence.yaml (before it has one key called 'workers')

* Compile aqua for 'fluence workers deploy' similar to what 'fluence deal deploy' produced

* Include peerIds of where workers where deployed using 'fluence workers deploy'

* Check for previously deployed stuff before uploading for both direct hosting and deals. And allow to select which ones you want to redeploy

### CODE REVIEW

File: src/commands/deal/deploy.ts [hunk 1]
```diff
@@ -180,31 +174,10 @@ export default class Deploy extends BaseCommand<typeof Deploy> {
...
-        mainAquaFileContent.replace(
-          MAIN_AQUA_FILE_STATUS_TEXT_COMMENT,
-          MAIN_AQUA_FILE_STATUS_TEXT
-        ),
-        FS_OPTIONS
-      );
-    } catch {}
-
+    await ensureAquaFileWithWorkerInfo(workersConfig);
     commandObj.log(`\nDeploy completed successfully`);
```
Comment (folex) [inline src/commands/deal/deploy.ts:181]: Would be nice to outline what has been deployed. Some kind of status.

File: src/commands/workers/deploy.ts [hunk 2]
```diff
@@ -96,20 +104,29 @@ export default class Deploy extends BaseCommand<typeof Deploy> {
...
+      },
+      { ...workersConfig.hosts }
     );
 
-    deployedConfig.workers = newDeployedWorkers;
-    await deployedConfig.$commit();
+    workersConfig.hosts = newDeployedWorkers;
+    await workersConfig.$commit();
+    await ensureAquaFileWithWorkerInfo(workersConfig);
     commandObj.log("Successfully deployed");
```
Comment (folex) [inline src/commands/workers/deploy.ts:130]: Ditto. Would be nice to outline what has been deployed. Some kind of status.

File: src/commands/workers/logs.ts [hunk 3]
```diff
@@ -164,14 +190,86 @@ const getLogsArg = async ({
...
+
   return {
-    workers: Object.entries(deployedConfig.workers)
+    workers: workers
       .filter(([name]) => workersToGetLogsFor.includes(name))
       .map(([name, config]) => ({ name, ...config })),
   };
 };
+
+const resolveDealsOrHosts = async (
```
Comment (folex) [inline src/commands/workers/logs.ts:216]: very nice!

File: src/lib/configs/project/workers.ts [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+  $id: `${TOP_LEVEL_SCHEMA_ID}/${WORKERS_CONFIG_FILE_NAME}`,
+  title: WORKERS_CONFIG_FILE_NAME,
+  type: "object",
+  description:
+    "A result of app deployment. This file is created automatically after successful deployment using `fluence workers deploy` command",
+  properties: {
+    version: { type: "number", const: 0 },
+    deals: {
+      type: "object",
+      description: "A map of deployed workers",
```
Comment (folex) [inline src/lib/configs/project/workers.ts:100]:
```suggestion
      description: "A map of created deals",
```

File: src/commands/deal/deploy.ts [hunk 5]
```diff
@@ -124,19 +119,9 @@ export default class Deploy extends BaseCommand<typeof Deploy> {
...
-        maybePreviouslyDeployedDeal.dealId !== undefined &&
-        (await confirm({
-          message: `There is a previously deployed deal for worker ${color.yellow(
-            workerName
-          )} on network ${color.yellow(
-            chainNetwork
-          )}. Do you want to update this existing deal?`,
-        }))
-      ) {
+      const maybePreviouslyDeployedDeal = workersConfig.deals?.[workerName];
```
[the remark this answers is not in the document]
Comment (shamsartem) [reply src/commands/deal/deploy.ts:122]: with ? it won't.
Thankfully with typescript config that we have I'v never had issues like this in this code base. Everything is sound

File: src/lib/configs/project/workers.ts [hunk 6]
```diff
@@ -0,0 +1,179 @@
...
+  }[];
+  timestamp: string;
+  definition: string;
+};
+
+const workerInfoSchema = {
+  type: "object",
+  properties: {
+    installation_spells: {
+      type: "array",
```
[the remark this answers is not in the document]
Comment (shamsartem) [reply src/lib/configs/project/workers.ts:50]: that's what we have in aqua - hence we have it here as well. I thought there is some bigger picture to it when I implemented it

File: src/commands/deal/deploy.ts (hunk 1 above)
Comment (shamsartem) [reply src/commands/deal/deploy.ts:181]: let's come up with something. What can it be? The one thing that I absolutely can show is the stuff that I write to the new workers.yaml (which previously was deployed.yaml)

Comment (folex) [reply src/commands/deal/deploy.ts:181]: > workers.yaml
Yeah, at least that, for now.



Later we will have to do some kind of monitoring for deployments. Like
```
Deploy completed successfully
Deal contract is: <color>0xabcacbacbacbacbacba</color>

Do you want to monitor providers accepting the deal? y/n
y
Waiting for providers to provide collateral... (100 USDC per a single worker)

12D3ABCACBACBACXYZ: 100 USDC
12D3UUUUUAAAAHHHHA: 200 USDC
12D3BYBYBYBYBYBYBY: 300 USDC
```
And then maybe even give a way to retrieve logs of Installation Spells of workers on these providers

**Review by folex — APPROVED** [review]
Good one!

File: src/lib/configs/project/workers.ts (hunk 6 above)
[the remark this answers is not in the document]
Comment (folex) [reply src/lib/configs/project/workers.ts:50]: https://github.com/fluencelabs/spell/blob/313d73b65cf8005933bb8965d5f2c9bd898c3b5f/src/aqua/installation-spell/src/aqua/config.aqua#L82

**Pull request merged**